[WNMGDS-1724] Update button doc page to reflect new button architecture (#1998)
CMSgov/design-system: packages/docs/content/components/button.mdx

  sketchLink: Vr4yyWw
---

import { ArrowIcon, Button } from '@cmsgov/design-system';

## Examples

> The following tables give an overview of the different permutations of buttons available in the selected design system. The controlling traits are the named _variation_, whether it's the _main_ vs. _alternate_ color, and the _on-light_ vs. _on-dark_ context. Each of these traits has its own example later on showing how to apply it.

Author: I'm unsure about using the word "permutations", but I can't think of a good alternative. Ideas? Ideas on making this section clearer in general?

Reviewer: Instead of "permutations", maybe something like "different available button styles"? Or update the paragraph a little like:

The following tables give an overview of the different available button styles in the selected design system. Each button has three controllable traits that can be used in various combinations. They are:
1. _Variation_: which affects the border and background styles
2. Color: defined as either _main_ or _alternate_
3. Context: specified as either _on-light_ or _on-dark_

See the below examples of how these are used.

Author: I like the rewrite!

Reviewer 2: "different permutations" -> "variations"

<ButtonVariationsTable />

> ### Outline buttons

Reviewer 3: ```suggestion
### Outline button
```

<StorybookExample
  componentName="button"
  sourceFilePath="components/Button/Button.tsx"
  storyId="components-button--default&args=children:Outline"
/>

> ### Solid buttons

Reviewer 3: ```suggestion
### Solid button
```

<StorybookExample

>   componentName="button"

Reviewer: Please update the `componentName` prop to be unique for each usage of `<StorybookExample>` because it is used to generate a unique ID and title for each iframe.

Author: Good to know, thanks!

  sourceFilePath="components/Button/Button.tsx"
  storyId="components-button--default&args=children:Solid;variation:solid"
/>

> ### Ghost buttons

Reviewer 3: ```suggestion
### Ghost button
```

<StorybookExample
  componentName="button"
  sourceFilePath="components/Button/Button.tsx"
  storyId="components-button--default&args=children:Ghost;variation:ghost"
/>

### Alternate buttons

> Alternate buttons can exist in each of the three main variations (outline, solid, ghost) but stand apart as having an alternate color. In the core design system, these follow the secondary brand color.

Reviewer 3: ```suggestion
Each button variation (outline, solid, ghost) has an alternate option. The alternate options can be styled differently than the main buttons.
```
